### DISCUSSION TITLE
ResizeWithPadOrCropd does not pad background channel in labels.

### DISCUSSION THREAD

**Discussion by umgpy** [Project-MONAI/MONAI#3908] [Q&A] [answered]
**Describe the bug**
`ResizeWithPadOrCropd` does not pad background channel for label data. 

**Expected behavior**
For a channel wise binary data, with background as 1, `ResizeWithPadOrCropd` should add 1's instead of 0's where padding takes place

**Screenshots**
![Screenshot from 2022-03-09 12-44-44](https://user-images.githubusercontent.com/44013803/157391324-913dc40a-d2ea-4ae6-b112-2b1619861276.png)

Top row is the ground truth label, with padding and bottom (predicted labels) is the expected padding behavior.

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
================================
Printing MONAI config...
================================
MONAI version: 0.7.0
Numpy version: 1.21.1
Pytorch version: 1.7.1+cu92
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: bfa054b9c3064628a21f4c35bbe3132964e91f43

Optional dependencies:
Pytorch Ignite version: 0.4.6
Nibabel version: 3.2.1
scikit-image version: 0.18.2
Pillow version: 8.3.1
Tensorboard version: 2.6.0
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.8.2+cu92
tqdm version: 4.62.3
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.
pandas version: 1.3.1
einops version: 0.3.2
transformers version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
`psutil` required for `print_system_info`

================================
Printing GPU config...
================================
Num GPUs: 4
Has CUDA: True
CUDA version: 9.2
cuDNN enabled: True
cuDNN version: 7603
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_70']
GPU 0 Name: Tesla V100-DGXS-32GB
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 80
GPU 0 Total memory (GB): 31.7
GPU 0 CUDA capability (maj.min): 7.0
```

**Additional context**
My code for transforms- 

```python
fn_keys = ["Img","Seg"]


val_transforms = Compose(
    [
        LoadImaged(keys=fn_keys),
        EnsureChannelFirstd(keys="Img"),
        ConvertToMultiChanneld(keys="Seg"),
        ResizeWithPadOrCropd(spatial_size=(256,512,512),keys=fn_keys),),
        EnsureTyped(keys=fn_keys),
    ]
)
```

**Comment by Nic-Ma** [comment]
Hi @umgpy ,

I think you can set the constant arg of `np.pad` as 1 for padding in the transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/dictionary.py#L1424

Thanks.

**Comment by wyli** [comment, marked as answer]
perhaps change the padding mode to `"edge"` https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/dictionary.py#L1416-L1417 (I'm converting this to a discussion)

**Reply by umgpy** [reply]
Hi, @Nic-Ma , @wyli 

Thank you for the quick replies!

Trying `constant_values` args doesn't work as it fills the value for all channels, as seen here-
 

![Screenshot from 2022-03-09 13-32-09](https://user-images.githubusercontent.com/44013803/157403036-e196ede9-272f-4ae5-82c9-f27aa5c5e039.png)

```python
ResizeWithPadOrCropd(spatial_size=(256,512,512),keys=fn_keys, constant_values = 1)
```


At the same time mode '"edge"' for the label does work channel wise but it has unintended consequence of elongating the label mask, which is not right.  

![Screenshot from 2022-03-09 14-00-06](https://user-images.githubusercontent.com/44013803/157403468-177cc6a3-8617-4f21-8ef2-87bec0d6284d.png)


```python
 ResizeWithPadOrCropd(spatial_size=(256,512,512),keys=fn_keys, mode = ['constant','edge'])
```

Thanks again for all the help!

**Reply by umgpy** [reply]
I also tried  `"median"` which works mostly

![Screenshot from 2022-03-09 14-21-48](https://user-images.githubusercontent.com/44013803/157406646-9b0afe8d-71f6-4083-9e88-a66db38ed9c8.png)

But at times there are artifacts, such as this-
![Screenshot from 2022-03-09 14-21-27](https://user-images.githubusercontent.com/44013803/157406670-7eb5d8e8-e6cd-4849-9942-6b1f225842da.png)

**Reply by wyli** [reply]
I see, perhaps you can do spatial padding before `ConvertToMultiChanneld`

**Reply by umgpy** [reply]
Thanks, this solved it!
